src/libraries/System.Private.CoreLib/src/System/MemoryExtensions.Trim.cs

```diff
@@ -565,27 +566,38 @@ public static ReadOnlyMemory<char> TrimEnd(this ReadOnlyMemory<char> memory)
         /// Removes all leading and trailing white-space characters from the span.
         /// </summary>
         /// <param name="span">The source span from which the characters are removed.</param>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public static ReadOnlySpan<char> Trim(this ReadOnlySpan<char> span)
         {
-            int start = 0;
-            for (; start < span.Length; start++)
+            // Assume that in most cases input doesn't need trimming
+            if (span.Length == 0 ||
+                (span.Length > 0 && !char.IsWhiteSpace(span[0]) && !char.IsWhiteSpace(span[^1])))
```

Comment:
```suggestion
                (!char.IsWhiteSpace(span[0]) && !char.IsWhiteSpace(span[^1])))
```
?

Reply:
Does the bounds check on the span[^1] get elided?

Reply:
> Does the bounds check on the span[^1] get elided?

No, but the codegen is the same as `span[span.Length - 1]`. As far as I remember we have an issue somewhere for bound check elimination for `Length - CNS`